File: questions/exam/autoR/avocadoHoustonPlot/tests/ans.R
library(tidyverse)

avocado <- read_csv("/grade/serverFilesCourse/dataR/avocado_prices.csv")

avocado_plot <- avocado |>
    mutate(total_volume = small_hass_volume + large_hass_volume + extra_l_hass_volume) |>
    filter(region == "Houston") |>
    ggplot(aes(x = total_volume, y = average_price)) + 
        geom_point() + 
        xlab("Total avocado volume sold \nin Houston (lbs)") +
        ylab("Average avocado price \nin Houston (USD)") +
        scale_x_log10() +
        theme(text = element_text(size = 20))

File: questions/exam/autoPy/avocadoHoustonPlot/tests/ans.py
import altair as alt
import pandas as pd

avocado = pd.read_csv("/grade/serverFilesCourse/dataR/avocado_prices.csv")

avocado = avocado.assign(
    total_volume=avocado["small_hass_volume"]
        + avocado["large_hass_volume"]
        + avocado["extra_l_hass_volume"]
)

avocado_plot = alt.Chart(avocado[avocado["region"] == "Houston"]).mark_point().encode(
    x=alt.X("total_volume")
        .title("Total avocado volume sold in Houston (lbs)")
        .scale(type="log"),
    y=alt.Y("average_price").title("Average avocado price in Houston (USD)"),
)